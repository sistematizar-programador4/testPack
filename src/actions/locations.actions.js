import { ADD_COORDS } from "./types";

export const addCoords = coords => ({
  type: ADD_COORDS,
  data: coords
});