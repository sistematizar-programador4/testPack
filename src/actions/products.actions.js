import { ADD_PRODUCT, DELETE_PRODUCT, EDIT_PRODUCT } from "./types";

export const addProduct = product => ({
  type: ADD_PRODUCT,
  data: product
});

export const editProduct = product => ({
  type: EDIT_PRODUCT,
  data: product
});

export const deleteProduct = key => ({
  type: DELETE_PRODUCT,
  key: key
});
