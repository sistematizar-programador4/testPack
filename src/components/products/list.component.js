import React, { Component } from "react";
import { StyleSheet, FlatList } from "react-native";
import { ListItem, Icon } from "react-native-elements";
import { connect } from "react-redux";
import { deleteProduct } from "../../actions/products.actions";

class ProductList extends Component {
  static navigationOptions = {
    title: "Products List",
    headerTintColor: "white",
    headerStyle: {
      backgroundColor: "#845cc3"
    }
  };

  render() {
    return (
      <FlatList
        style={styles.listContainer}
        data={this.props.products}
        keyExtractor={(item, index) => item.key.toString()}
        renderItem={data => (
          <ListItem
            title={data.item.name}
            input={{}}
            leftIcon={
              <Icon
                name="edit"
                size={18}
                onPress={() => this.props.navigation.navigate("ProductForm", {
                  product: data.item.name,
                  key: data.item.key
                })}
              />
            }
            rightIcon={
              <Icon
                name="delete"
                size={18}
                onPress={() => this.props.delete(data.item.key)}
              />
            }
          />
        )}
      />
    );
  }
}

const styles = StyleSheet.create({
  listContainer: {
    backgroundColor: "#dce2ff",
    padding: 16
  },
  listText: {
    fontSize: 30
  }
});

const mapStateToProps = state => {
  console.log(state);
  return {
    products: state.productReducer.productList
  };
};

const mapDispatchToProps = dispatch => {
  return {
    delete: key => dispatch(deleteProduct(key))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ProductList);
