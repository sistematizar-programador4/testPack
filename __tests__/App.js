import 'react-native';
import React from 'react';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import * as products from '../src/actions/products.actions';
import * as locations from '../src/actions/locations.actions';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const middlewares = [thunk]
const mockStore = configureStore(middlewares)
const store = mockStore({})

beforeEach(() => {
  store.clearActions();
});

it('ADD_PRODUCT', () => {
  store.dispatch(products.addProduct({
    key: Math.random(),
    name: 'espinaca'}
  ));
  expect(store.getActions()).toMatchSnapshot();
});

it('DELETE_PRODUCT', () => {
  store.dispatch(products.deleteProduct(0.5880098879062283));
  expect(store.getActions()).toMatchSnapshot();
});

it('ADD_COORDS', () => {
  store.dispatch(locations.addCoords({
    key: Math.random(),
    location: '00000'}
  ));
  expect(store.getActions()).toMatchSnapshot();
});
