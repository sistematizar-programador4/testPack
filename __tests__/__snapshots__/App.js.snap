// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADD_COORDS 1`] = `
Array [
  Object {
    "data": Object {
      "key": 0.05723000657152855,
      "location": "00000",
    },
    "type": "ADD_COORDS",
  },
]
`;

exports[`ADD_PRODUCT 1`] = `
Array [
  Object {
    "data": Object {
      "key": 0.9008418350000806,
      "name": "espinaca",
    },
    "type": "ADD_PRODUCT",
  },
]
`;

exports[`DELETE_PRODUCT 1`] = `
Array [
  Object {
    "key": 0.5880098879062283,
    "type": "DELETE_PRODUCT",
  },
]
`;
